qiskit/circuit/qpy_serialization.py

```diff
@@ -1220,6 +1317,32 @@ def _write_instruction(file_obj, instruction_tuple, custom_instructions, index_m
             container.seek(0)
             data = container.read()
             size = len(data)
+        elif param is None:
+            type_key = "z"
+            data = b""
+            size = len(data)
+        elif isinstance(param, QuantumCircuit):
+            type_key = "q"
+            _write_circuit(container, param)
+            data = container.getvalue()
+            size = len(data)
+        elif isinstance(param, range):
+            type_key = "r"
+            data = struct.pack(RANGE_PACK, param.start, param.stop, param.step)
+            size = len(data)
+        # Tuples are arbitrary containers of python objects and can have anythin
+        # as a value, qpy doesn't support this as a type currently. But in the
+        # ForLoopOp the indexset parameter is a tuple of only intergers
```

Comment:
```suggestion
        # Tuples are arbitrary containers of python objects and can have anything
        # as a value, qpy doesn't support this as a type currently. But in the
        # ForLoopOp the indexset parameter is a tuple of only integers
```

Ah - I made a comment previously about whether it's worth doing this immediately.  It feels like it'd be easier to maintain the code later if we do it now, but I've not got a huge amount of experience in YAGNI policy...